mpas_analysis/shared/timekeeping/Date.py

```diff
@@ -75,6 +76,50 @@ def __init__(self, dateString=None, isInterval=False, totalSeconds=None,
             self.seconds = numpy.int64(seconds)
             self._setTotalSeconds()
 
+    def to_datetime(self, yearOffset=0):
+        """
+        Converts the date object to a datetime object.
+        The yearOffset is added to this date's year, and
+        the resulting date is clamped to the range supported by
+        numpy's datetime64[ns], used internally by xarray an
+        pandas
+
+        Last modified: 11/28/2016
+        Author: Xylar Asay-Davis
+        """
+        if self.isInterval:
+            raise ValueError("self.isInterval == True. Use to_timedelta "
+                             "instead of to_datetime")
+
+        year = numpy.maximum(datetime.MINYEAR,
+                             numpy.minimum(datetime.MAXYEAR,
+                                           self.years+yearOffset))
+        outDate =  datetime.datetime(year=year, month=self.months+1,
+                                     day=self.days+1, hour=self.hours,
+                                     minute=self.minutes, second=self.seconds)
+
+        minDate = datetime.datetime(year=1678, month=1, day=1,
+                                    hour=0, minute=0, second=0)
+        maxDate = datetime.datetime(year=2262, month=1, day=1,
+                                    hour=0, minute=0, second=0)
+        outDate = max(minDate, min(maxDate, outDate))
```

Comment:
@pwolfram, we don't support going past 100 years right now (because of limited numbers of files) so I think 2262 is the least of our concerns...

But also fixing `Date` with a new min/max allowed date will be super easy once xarray and pandas support a wider range.  For now, we're at their mercy and this function is really only a side effect of that.